File: sorting/python/SelectSort.py
# -*- coding: utf-8 -*
'''
    选择排序
'''
def selectSort(arr):
    n = len(arr)
    for i in range(0,n,1):
        minIndex = i;
        #找到最小的一个值
        for j in range(i+1,n,1):
            if arr[j] < arr[minIndex]:
                minIndex = j

        # sawp 操作 交换位置
        temp = arr[i]
        arr[i] = arr[minIndex]
        arr[minIndex] = temp

if __name__ == '__main__':
    arr = [3, 4, 2, 5, 8, 6, 9, 1]
    selectSort(arr)
    for i in arr:
        print(i)

File: sorting/golang/select_sort.go
package main

import (
	"./help"
)

func selectSort(arr []int, n int) []int {
	for i := 0; i < n; i++ {
		minIndex := i
		// 遍历出最小的索引位置
		for j := i + 1; j < n; j++ {
			if arr[minIndex] > arr[j] {
				minIndex = j
			}
		}
		// swap
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}

	return arr
}

func main() {
	arr := help.GenRanArray(10000, 1, 9999)
	// help.ArrayPrint(arr)
	help.TestSort(selectSort)(arr, len(arr))
}
